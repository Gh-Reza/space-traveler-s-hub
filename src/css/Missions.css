.missions-table {
  width: 90%;
  margin-left: 3rem;
}

table,
th,
td {
  text-align: left;
  padding: 0.5rem;
  border-collapse: collapse;
  border: 1px solid rgb(185, 184, 184);
}

th {
  font-size: 1.2rem;
}

.description-column {
  width: 62%;
}

.status-column {
  width: 13%;
}

.button-column {
  width: 12%;
}

.join-btn,
.leave-btn {
  height: 2rem;
  padding: 0.5rem;
  border-radius: 8%;
  border-width: 2px;
}

.leave-btn {
  border-color: rgb(214, 76, 76);
  color: rgb(214, 76, 76);
}

table tr:nth-child(odd) td {
  background-color: rgb(239, 237, 237);
}

.status-active,
.status-not-active {
  padding: 0.3rem;
  border-radius: 0.5rem;
  color: white;
  font-size: 0.9rem;
}

.status-not-active {
  background-color: rgb(151, 149, 149);
}

.status-active {
  background-color: rgb(62, 142, 148);
}

.mission-description {
  font-size: 0.9rem;
  line-height: normal;
}

.mission-name {
  font-weight: bold;
}
