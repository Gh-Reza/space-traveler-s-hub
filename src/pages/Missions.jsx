import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { fetchMissions } from '../redux/missions/miissionsSlice';

const Missions = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);
  const missions = useSelector((state) => state.missions);
    <div>
      <h2>
        Missions
      </h2>
      <p>{missions}</p>
    </div>;
};

export default Missions;
