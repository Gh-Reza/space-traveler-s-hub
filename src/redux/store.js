import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsslice';
import missionsReducer from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketReducer,
  },
}, applyMiddleware);

export default store;
