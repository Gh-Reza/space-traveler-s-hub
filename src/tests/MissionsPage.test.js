import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { screen, render } from '@testing-library/react';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Missions from '../pages/Missions';

describe('Mission Page unit test', () => {
  test('Mission Page Renders Properly ', () => {
    const tree = renderer.create(<Provider store={store}><Missions /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('The Mission Page Contains Mission Table Headers', () => {
    render(<Provider store={store}><Missions /></Provider>);
    expect(screen.getByText('Mission')).toBeInTheDocument();
    expect(screen.getByText('Description')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
  });
});
