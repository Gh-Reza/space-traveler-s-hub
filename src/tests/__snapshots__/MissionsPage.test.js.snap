// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Page unit test Mission Page Renders Properly  1`] = `
<div>
  <table
    className="missions-table"
  >
    <tr>
      <th>
        Mission
      </th>
      <th
        className="description-column"
      >
        Description
      </th>
      <th
        className="status-column"
      >
        Status
      </th>
      <th
        className="button-column"
      >
          
      </th>
    </tr>
    ;
  </table>
  <p />
</div>
`;
